tests/test_algebra_onnx_operators_sub_estimator.py

```diff
@@ -0,0 +1,148 @@
+
+
+class ValidatorClassifier(BaseEstimator, ClassifierMixin):
+
+    def __init__(self, estimator=None, threshold=0.75):
+        ClassifierMixin.__init__(self)
+        BaseEstimator.__init__(self)
+        if estimator is None:
+            estimator = LogisticRegression(solver='liblinear')
+        self.estimator = estimator
+        self.threshold = threshold
+
+    def fit(self, X, y, sample_weight=None):
+        sig = inspect.signature(self.estimator.fit)
+        if 'sample_weight' in sig.parameters:
+            self.estimator_ = clone(self.estimator).fit(
+                X, y, sample_weight=sample_weight)
+        else:
+            self.estimator_ = clone(self.estimator).fit(X, y)
+        return self
+
+    def predict(self, X):
+        return self.estimator_.predict(X)
+
+    def predict_proba(self, X):
+        return self.estimator_.predict_proba(X)
+
+    def validate(self, X):
+        pred = self.predict_proba(X)
+        mx = pred.max(axis=1)
+        return (mx >= self.threshold) * 1
+
+
+def validator_classifier_shape_calculator(operator):
+
+    input = operator.inputs[0]  # inputs in ONNX graph
+    outputs = operator.outputs  # outputs in ONNX graph
+    op = operator.raw_operator  # scikit-learn model (mmust be fitted)
+    if len(outputs) != 3:
+        raise RuntimeError("3 outputs expected not {}.".format(len(outputs)))
+
+    N = input.type.shape[0]                 # number of observations
+    C = op.estimator_.classes_.shape[0]     # dimension of outputs
+
+    outputs[0].type = Int64TensorType([N])     # label
+    outputs[1].type = FloatTensorType([N, C])  # probabilities
+    outputs[2].type = Int64TensorType([C])     # validation
+
+
+def validator_classifier_converter(scope, operator, container):
+    input = operator.inputs[0]      # input in ONNX graph
+    outputs = operator.outputs      # outputs in ONNX graph
+    op = operator.raw_operator      # scikit-learn model (mmust be fitted)
+    opv = opv = container.target_opset
+
+    # We reuse existing converter and declare it as local
+    # operator.
+    model = op.estimator_
+    onnx_op = OnnxSubEstimator(model, input, op_version=opv)
+
+    rmax = OnnxReduceMax(onnx_op[1], axes=[1], keepdims=0, op_version=opv)
+    great = OnnxGreater(rmax, np.array([op.threshold], dtype=np.float32),
+                        op_version=opv)
+    valid = OnnxCast(great, to=onnx_proto.TensorProto.INT64,
+                     op_version=opv)
+
+    r1 = OnnxIdentity(onnx_op[0], output_names=[outputs[0].full_name],
+                      op_version=opv)
+    r2 = OnnxIdentity(onnx_op[1], output_names=[outputs[1].full_name],
+                      op_version=opv)
+    r3 = OnnxIdentity(valid, output_names=[outputs[2].full_name],
+                      op_version=opv)
+
+    r1.add_to(scope, container)
+    r2.add_to(scope, container)
+    r3.add_to(scope, container)
+
+
+def validator_classifier_parser(scope, model, inputs, custom_parsers=None):
```

Comment:
Unused argument custom_parsers

Reply:
All other parsers follow that signature. It is used for sub models.